api/api-rules/violation_exceptions.list

```diff
@@ -61,6 +61,7 @@ API rule violation: names_match,k8s.io/kube-controller-manager/config/v1alpha1,A
 API rule violation: names_match,k8s.io/kube-controller-manager/config/v1alpha1,CSRSigningControllerConfiguration,ClusterSigningCertFile
 API rule violation: names_match,k8s.io/kube-controller-manager/config/v1alpha1,CSRSigningControllerConfiguration,ClusterSigningDuration
 API rule violation: names_match,k8s.io/kube-controller-manager/config/v1alpha1,CSRSigningControllerConfiguration,ClusterSigningKeyFile
+API rule violation: names_match,k8s.io/kube-controller-manager/config/v1alpha1,CronJobControllerConfiguration,CronJobControllerSyncPeriod
```

Comment:
Hmm I made this change since I saw it done [here](https://github.com/kubernetes/kubernetes/pull/66840/commits/cbdc9b671f33b0f0679e790cc462b25d1476a3af#diff-cb775184a18e11181dfe385defc37237R129) without fully understanding what this does. Let me remove it and run the tests locally to see what it does

Reply:
Looking back on when I hit it, it may be a necessity for adding config APIs? I last hit it in https://github.com/kubernetes/kubernetes/pull/79576/files#diff-cb775184a18e11181dfe385defc37237R478 

Some names that I had to list, like `ClusterCIDR`, are violations. Others, like KubeProxyConfiguration, don't violate the naming policy.

Reply:
So when I run `make generated_files`, I get:

```
66d65
< API rule violation: names_match,k8s.io/kube-controller-manager/config/v1alpha1,CronJobControllerConfiguration,CronJobControllerSyncPeriod
110d108
< API rule violation: names_match,k8s.io/kube-controller-manager/config/v1alpha1,KubeControllerManagerConfiguration,CronJobController
ERROR:
         API rule check failed. Reported violations differ from known violations. Please read api/api-rules/README.md to resolve the failure.
```

Following https://github.com/kubernetes/kubernetes/tree/master/api/api-rules#how-to-resolve-api-rule-check-failure, I run `make generated_files UPDATE_API_KNOWN_VIOLATIONS=true` and it succeeds.

Reply:
Thanks for indulging me - I'll take my curiosity about _why_ this is a violation upstream.